File: Data Structures & Algorithms/search-2d-matrix/submission-3.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:




        for row in matrix:

            if target > max(row):
                continue
            elif target == max(row) or target == min(row):
                return True
            else:
                l=0
                r= len(row)-1
                
                while l<= r:

                    mid = l+ (r-l) //2

                    if row[mid]== target:
                        return True
                    elif target < row[mid]:
                        r= mid -1
                    elif target > row[mid]:
                        l= mid + 1
        return False


            


        

File: Data Structures & Algorithms/max-water-container/submission-3.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {

    
    int area = 0;



    int l = 0;
    int r = heights.size() -1;


    while ( l < r){

        int sum = min(heights[l], heights[r]) * (r - l);
        area = max(area,sum);

        if(heights[l] < heights[r] ){


            l++;
        }

        else{

            r--;
        }

        
    }

    return area;
    }
};
